sycl/test-e2e/ESIMD/fma.cpp

```diff
@@ -0,0 +1,112 @@
+//==---------------- fma.cpp - DPC++ ESIMD on-device test -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// RUN: %{build} -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %{run} %t.out
+
+#include "esimd_test_utils.hpp"
+
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/intel/experimental/esimd/math.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+template <typename T> bool test(queue q) {
+  std::cout << "Running case: T=" << esimd_test::type_name<T>() << std::endl;
+  constexpr unsigned Size = 16;
+  constexpr unsigned VL = 16;
+
+  T *A = new T[Size];
+  T *B = new T[Size];
+  T *C = new T[Size];
+
+  for (unsigned i = 0; i < Size; ++i) {
+    A[i] = B[i] = i;
+    C[i] = 0.0f;
+  }
+
+  try {
+    buffer<T, 1> bufa(A, range<1>(Size));
+    buffer<T, 1> bufb(B, range<1>(Size));
+    buffer<T, 1> bufc(C, range<1>(Size));
+
+    range<1> GlobalRange{Size / VL};
+
+    range<1> LocalRange{1};
+
+    auto e = q.submit([&](handler &cgh) {
+      auto PA = bufa.template get_access<access::mode::read>(cgh);
+      auto PB = bufb.template get_access<access::mode::read>(cgh);
+      auto PC = bufc.template get_access<access::mode::write>(cgh);
+      cgh.parallel_for(
+          GlobalRange * LocalRange, [=](id<1> i) SYCL_ESIMD_KERNEL {
+            using namespace sycl::ext::intel::esimd;
+            unsigned int offset = i * VL * sizeof(T);
+            simd<T, VL> va;
+            va.copy_from(PA, offset);
+            simd<T, VL> vb;
+            vb.copy_from(PB, offset);
+            simd<T, VL> vc = 5.0f;
+            simd<T, VL> vd = ext::intel::experimental::esimd::fma(va, vb, vc);
+            vd.copy_to(PC, offset);
+          });
+    });
+    e.wait();
```

Comment:
minor comment: why bother using buffers and accessors while they are still converted to stateless memory accesses?

Also, if Size=16 and VL=16, then this could be a 'single_task' instead of 'parallel_for'.
```suggestion
    q.parallel_for(
          GlobalRange * LocalRange, [=](id<1> i) SYCL_ESIMD_KERNEL {
          using namespace sycl::ext::intel::esimd;
          unsigned int offset = i * VL;
          simd<T, VL> va(PA + offset);
          simd<T, VL> vb(PB + offset);
          simd<T, VL> vc = 5.0f;
          simd<T, VL> vd = ext::intel::experimental::esimd::fma(va, vb, vc);
          vd.copy_to(PC + offset);
    }).wait();
```

Reply:
i based this test on vadd_1d which was written that way (i changed the size though), i'm going to merge it as-is if that's okay

Reply:
i based this test on vadd_1d which was written that way (i changed the size though), ill change the `single_task` since i have to rebase anyway but i'll leave it using accessors if that's okay